anitya/tests/test_sar.py

```diff
@@ -0,0 +1,130 @@
+# of Red Hat, Inc.
+#
+
+'''
+anitya tests for GDPR SAR script.
+'''
+
+import pytest
+
+import anitya.sar as sar
+from anitya.db import models
+from anitya.tests.base import DatabaseTestCase
+
+
+class SARTests(DatabaseTestCase):
+    """ SAR script tests. """
+
+    @pytest.fixture(autouse=True)
+    def capsys(self, capsys):
+        """ Use capsys fixture as part of this class. """
+        self.capsys = capsys
+
+    def setUp(self):
+        """ Prepare variables for tests. """
+        super(SARTests, self).setUp()
+
+        sar.SAR_EMAIL = ''
+        sar.SAR_USERNAME = ''
+
+    def test_main_email(self):
+        """
+        Assert that correct user data are dumped when providing
+        e-mail.
+        """
+        user = models.User(
+            email='user@fedoraproject.org',
+            username='user',
+            active=True,
+        )
+        self.session.add(user)
+
+        user2 = models.User(
+            email='user2@email.org',
+            username='user2',
+            active=True,
+        )
+        self.session.add(user2)
+        self.session.commit()
+
+        sar.SAR_EMAIL = 'user@fedoraproject.org'
+        sar.main()
+
+        out, err = self.capsys.readouterr()
+
+        self.assertTrue('"username": "{}"'.format(user.username) in out)
+        self.assertTrue('"email": "{}"'.format(user.email) in out)
+
+    def test_main_username(self):
+        """
+        Assert that correct user data are dumped when providing
+        username.
+        """
+        user = models.User(
+            email='user@fedoraproject.org',
+            username='user',
+            active=True,
+        )
+        self.session.add(user)
+
+        user2 = models.User(
+            email='user2@email.org',
+            username='user2',
+            active=True,
+        )
+        self.session.add(user2)
+        self.session.commit()
+
+        sar.SAR_USERNAME = 'user'
+        sar.main()
+
+        out, err = self.capsys.readouterr()
+
+        self.assertTrue('"username": "{}"'.format(user.username) in out)
+        self.assertTrue('"email": "{}"'.format(user.email) in out)
+        self.assertFalse('"username": "{}"'.format(user2.username) in out)
+        self.assertFalse('"email": "{}"'.format(user2.email) in out)
+
+    def test_main_no_env_set(self):
+        """
+        Assert that correct user data are dumped when nothing is provided.
+        """
+        user = models.User(
+            email='user@fedoraproject.org',
+            username='user',
+            active=True,
+        )
+        self.session.add(user)
+
+        user2 = models.User(
+            email='user2@email.org',
+            username='user2',
+            active=True,
+        )
+        self.session.add(user2)
+        self.session.commit()
+
+        sar.main()
+
+        out, err = self.capsys.readouterr()
+        print(out)
+
+        self.assertTrue('' in out)
```

Comment:
This will be ```True``` for any ```str```:

```
>>> '' in 'hello'
True
```

I recommend asserting that the output is exactly the JSON string that is expected (would that be ```[]```?)

Reply:
Thanks for noticing. I will change this to work with json.